### PR TITLE
Generate error message with conflicting Qt macros

### PR DESCRIPTION
**Pull request by philm001**
Hello all,

Here is the pull request related to issue #7524 . In this, I created a build error to appear if "signals" is already defined. the message also lets the user know of a possible fix. 

On a side note, I tried to update my fork to the latest version but when I went to creating a pull request, it states that the branch can be auto merged.

If you have any feedback, please let me know.

Best

### CODE REVIEW

**Comment by masterleinad** [comment]
It looks likely the indentation is off:
```
@@ -47,7 +47,8 @@
 DEAL_II_NAMESPACE_OPEN
 
 #ifdef signals
-#error "The data type signals is already defined. You are most likely using the QT library \
+#  error \
+    "The data type signals is already defined. You are most likely using the QT library \
 and using the signals keyword. You can either #include the Qt headers (or any conflicting headers) \
 after the deal.ii headers or you can define the QT_NO_KEYWORDS marco and use the Q_SIGNALS macro."
 #endif
diff --git a/include/deal.II/multigrid/multigrid.h b/include/deal.II/multigrid/multigrid.h
index 628da61d..07bbe2d3 100644
--- a/include/deal.II/multigrid/multigrid.h
+++ b/include/deal.II/multigrid/multigrid.h
@@ -37,7 +37,8 @@
 DEAL_II_NAMESPACE_OPEN
 
 #ifdef signals
-#error "The data type signals is already defined. You are most likely using the QT library \
+#  error \
+    "The data type signals is already defined. You are most likely using the QT library \
 and using the signals keyword. You can either #include the Qt headers (or any conflicting headers) \
 after the deal.ii headers or you can define the QT_NO_KEYWORDS marco and use the Q_SIGNALS macro."
 #endif
##[error]Process completed with exit code 1.
```
Can you just run the indentation script `contrib/utilities/indent` (or run `make indent` in your build directory)? Other than that, this looks OK to me.

**Comment by philm001** [comment]
Hello all,

I think that I have some settings incorrect. When I go to run the indent script, I get error saying "invalid author 'Phillip' without firstname and lastname"

I checked my git config:

user.name=Phillip Mobley
user.firstname=Phillip
user.lastname=Mobley
user.email=phillipmobley2@gmail.com

I am not which setting is incorrect but does it look like these settings are correct? If not, what is the correct setting that I should be editing?

**Comment by masterleinad** [comment]
> I am not which setting is incorrect but does it look like these settings are correct? If not, what is the correct setting that I should be editing?

You (most likely) need to modify the author information for the commits in this pull request. Have a look at https://www.git-tower.com/learn/git/faq/change-author-name-email in "Editing the Author of Past Commits" for how to do that.

**Comment by masterleinad** [comment]
The author informations for your last commits are
```
phillip <you@example.com>
```
You need to fix that. :slightly_smiling_face:

File: include/deal.II/grid/tria.h.orig [hunk 1]
```diff
@@ -0,0 +1,4283 @@
+// ---------------------------------------------------------------------
+//
```
Comment (masterleinad) [inline include/deal.II/grid/tria.h.orig:2]: I am pretty sure you didn't mean to add this file...

File: include/deal.II/multigrid/multigrid.h.orig [hunk 2]
```diff
@@ -0,0 +1,940 @@
+// ---------------------------------------------------------------------
+//
```
Comment (masterleinad) [inline include/deal.II/multigrid/multigrid.h.orig:2]: ... or this file.

Comment (philm001) [reply include/deal.II/multigrid/multigrid.h.orig:2]: That is interesting, not sure how that got in there

Comment (philm001) [reply include/deal.II/multigrid/multigrid.h.orig:2]: If I am reading that correctly, isn't that the first two lines in multigrid.h? Same for tria.h?

**Comment by philm001** [comment]
Ok, so in this commit, I ran:

`git commit --amen --author="Phillip Mobley<phillipmobley2@gmail.com>"`

However, when I run the `git config --list`, I do not see my name or the new email address. I know that the name is correct as you see it on your end. Not so sure about the email address. At any rate let me know if it updated.

**Comment by masterleinad** [comment]
> At any rate let me know if it updated.

No. I would just squash everything in one commit via 
```
git rebase -i upstream/develop
```
and marking everything but the first commit as "fixup". Make sure that you set your author information globally appropriately, though.

File: include/deal.II/multigrid/multigrid.h.orig (hunk 2 above)
Comment (philm001) [reply include/deal.II/multigrid/multigrid.h.orig:2]: Nevermind, I found this issue and the files are deleted

**Comment by philm001** [comment]
Still working on that pesky error  "invalid author 'Phillip' without firstname and lastname"

I checked my global settings and they seem fine. I have been running the git rebase command a number of times, I think I am missing something because it appears that the author settings have not been fixed.... I am still working on it

**Comment by masterleinad** [comment]
You need to make sure that all the commits you have here have correct author information. Since the actual changes in this pull request are not that big I would just reset to the last commit you branched off from via
```
git reset <commit>
```
If  your branch is rebased on `upstream/master`, you wpuld just do
```
git reset upstream/master
```
and then commit your relevant changes again. Don't forget that you need to force-push your branch then.

**Comment by philm001** [comment]
Ok, I think that I got it now

**Comment by masterleinad** [comment]
Hmmm... Now, your commit message is "Fixed issue related to authorship" which doesn't really describe the changes. Would you min rewording? Just do a `git commit --amend`, fix the message and force-push again.

**Comment by philm001** [comment]
There we go. Quick fix

**Comment by philm001** [comment]
Great thanks, yeah, it must be a rite of passage!

**Comment by masterleinad** [comment]
Why did you close your pull request?

**Comment by masterleinad** [comment]
I am pretty sure that wasn't intentional so I reopened.

**Comment by philm001** [comment]
(－‸ლ)

**Pull request merged**